libctru/include/3ds/services/ndm.h

```diff
@@ -4,21 +4,124 @@
  */
 #pragma once
 
+/// Exclusive states.
 typedef enum {
 	EXCLUSIVE_STATE_NONE = 0,
 	EXCLUSIVE_STATE_INFRASTRUCTURE = 1,
 	EXCLUSIVE_STATE_LOCAL_COMMUNICATIONS = 2,
 	EXCLUSIVE_STATE_STREETPASS = 3,
 	EXCLUSIVE_STATE_STREETPASS_DATA = 4,
-} NDM_ExclusiveState;
+} ndmExclusiveState;
+
+/// Current states.
+typedef enum {
+	STATE_INITIAL = 0,
+	STATE_SUSPENDED = 1,
+	STATE_INFRASTRUCTURE_CONNECTING = 2,
+	STATE_INFRASTRUCTURE_CONNECTED = 3,
+	STATE_INFRASTRUCTURE_WORKING = 4,
+	STATE_INFRASTRUCTURE_SUSPENDING = 5,
+	STATE_INFRASTRUCTURE_FORCE_SUSPENDING = 6,
+	STATE_INFRASTRUCTURE_DISCONNECTING = 7,
+	STATE_INFRASTRUCTURE_FORCE_DISCONNECTING = 8,
+	STATE_CEC_WORKING = 9,
+	STATE_CEC_FORCE_SUSPENDING = 10,
+	STATE_CEC_SUSPENDING = 11,
+} ndmState;
+
+// Daemons.
+typedef enum {
+	DAEMON_CEC = 0,
+	DAEMON_BOSS = 1,
+	DAEMON_NIM = 2,
+	DAEMON_FRIENDS = 3,
+} ndmDaemon;
+
+/// Used to specify multiple daemons.
+typedef enum {
+	DAEMON_MASK_CEC = 1 << DAEMON_CEC,
```

Comment:
I would use `DAEMON_MASK_CEC = BIT(DAEMON_CEC)` and similar for next lines. I would probably change the name to `DAEMON_CEC_BIT`.

Reply:
Just to make sure, you want to DAEMON_CEC -> DAEMON_CEC_BIT?

Reply:
Logically I suppose either way could be argued. I might prefer `DAEMON_CEC_BIT = 0` and `DAEMON_CEC = BIT(DAEMON_CEC_BIT)`